cnnimg.py

```diff
@@ -65,111 +62,107 @@ def imageprocessing(self, inputlocation, outputlocation, tempA, tempB, initialco
         u = np.array(gray)
         u = u[:,:,0]
         z0 = (u)*initialcondition
-        self.Bu = sig.convolve2d(u, tempB, 'same')
-        self.tempA = tempA
-        self.Ib = Ib
-        z0 = z0.reshape(z0.size)
-        z = self.cnn(sint.odeint(self.f, z0, t))
+        Bu = sig.convolve2d(u, tempB, 'same')
+        z0 = z0.flatten()
+        z = self.cnn(sint.odeint(self.f, z0, t, args=(Ib, Bu, tempA)))
         l = z[z.shape[0]-1,:].reshape((self.n, self.m))
         l = l/(255.0)
-        for i in range(l.shape[0]):
-            for j in range(l.shape[1]):
-                l[i][j] = np.uint8(round(l[i][j]*255))
+        l = np.uint8(np.round(l*255))
         l = img.fromarray(l).convert('RGB')
         l.save(outputlocation)
-        print "Image Processing is successful."
+        print("Image Processing is successful.")
         return
     ## Helper Functions end
 
-    ## Image Processing methods start    
+    ## Image Processing methods start
 
     #edge detection #working
     def edgedetection(self, inputlocation = "", outputlocation = "output.png"):
         if not self.isvalid(inputlocation):
-            print "Invalid Location. Please try again ... "
+            print ("Invalid Location. Please try again ... ")
             exit()
-        print "Edge detection is initialized ... "
-        tempA = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.0]])
+        print ("Edge detection is initialized ... ")
+        tempA = np.zeros((3,3), dtype=float)
```

Comment:
The array representation should be consistent. That's why I wanted to avoid this representation.